### ISSUE TITLE
Upgrading to v5.1.0 jqm es6 Object.assign(..)

### ISSUE THREAD

**Issue by gregh3269** [jtsage/jtsage-datebox#473]
Loading datebox for jqm crashes the page on an older webkit phone, there is some es6 Object.assign(..) code.

I did some testing and the suggested polyfill 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill

works.
Should fail gracefully.  ...Alas, the work of js is moving to fast (or lack of foresight) for reality.

**Comment by jtsage** [comment]
Gross.  I will drop the polyfill in the jqm build - because chances are the other builds are also broken, but I bet those frameworks aren't going to work either without a third party shim on devices that are that old.

**Comment by gregh3269** [comment]
Down loaded v5.1.3 and should the fix be inside jtsage-datebox.min.js & jtsage-datebox.js?

**Comment by jtsage** [comment]
Yes, yes it should.  But it's not, due to how the build process works.  It'd be there if node didn't have Object.assign.  I'll look at this today

**Comment by jtsage** [comment]
So, as part of this, I'm adding a folder, [root]/src/js/polyfill - right now it's only used for jQM, but it could be expanded pretty easily.  Pretty much, drop a .js file in that folder and it will be included in the jQM build.  The Object.assign bit is in there - I don't believe I use any other "modern" javascript bits - I actually hesitated over this one for a bit, as most of the devices that don't have it have gone out of service - or so I thought anyway :)

I'm gonna push to the main server and build the -latest branch (http://cdn.jtsage.com/jtsage-datebox/latest/) if you want to verify this works before I tag the release.

**Comment by gregh3269** [comment]
Build seems OK, thanks.

**Comment by jtsage** [comment]
Great, I'll get this tagged and pushed to v5.1.4 sometime today.  Thanks!

**Comment by gregh3269** [comment]
I have tried 5.1.4 and the polyfill is still missing.  

looking here http://cdn.jtsage.com/jtsage-datebox/latest/jtsage-datebox.jqm.js I can see it.

Also, every time datebox fails it crashes the js on the page (blocking the page totally), could a try catch block be used somewhere to stop this? or maybe initialise when the button is clicked which would make page load faster?

**Comment by gregh3269** [comment]
Using the download builder v5.3.3 for jqm the polyfill is still missing.  Although it is present in the cdn link above.  Otherwise it works great.  Thanks.

**Comment by jtsage** [comment]
Oh, gross - because I re-wrote the build script for the cdn builds, but never touched the download builder.  Cool, I'll get this done as soon as I have a few minutes to sit with it.
